src/DeviceInterface/InterfacePrototype.js

```diff
@@ -610,6 +611,48 @@ class InterfacePrototype {
             this.storeFormData(withAutoGeneratedFlag)
         }
     }
+    /**
+     * on macOS we try to detect if a click occurred within a form
+     * @param {PointerEvent} event
+     */
+    _onPointerDown (event) {
+        if (this.settings.featureToggles.credentials_saving) {
+            this._detectFormSubmission(event)
+        }
+    }
+    /**
+     * @param {PointerEvent} event
+     */
+    _detectFormSubmission (event) {
+        const matchingForm = [...this.scanner.forms.values()].find(
+            (form) => {
+                const btns = [...form.submitButtons]
+                // @ts-ignore
+                if (btns.includes(event.target)) return true
+
+                // @ts-ignore
+                if (btns.find((btn) => btn.contains(event.target))) return true
+            }
+        )
+
+        matchingForm?.submitHandler()
+
+        if (!matchingForm) {
```

Comment:
This is new. If we couldn't find a matching form, I am now trying to read the clicked element to see if it's a button (or a descendant), then I check if it contains relevant text for login/signup stuff and check that the text is rather short so it's not a catchall. Finally, I see if there's a filled form anywhere in the page so that we can read that and prompt to store data. This fixes Google and Twitter form submission detection (and others, of course).